src/modules/rolle/domain/rolle.ts

```diff
@@ -174,10 +174,71 @@ export class Rolle<WasPersisted extends boolean> {
+        if (missingIds.length > 0) {
+            return new EntityNotFoundError('ServiceProvider', missingIds.join(', '));
+        }
+
+        const uniqueIds: Set<string> = new Set(serviceProviderIds);
+
+        // Filter out any IDs that are already in this.serviceProviderIds to avoid duplicates.
+        const newIds: string[] = [...uniqueIds].filter((id: string) => !this.serviceProviderIds.includes(id));
+
+        // Attach the new unique IDs
+        this.serviceProviderIds.push(...newIds);
+    }
+
+    public detatchServiceProvider(serviceProviderIds: string[]): void | DomainError {
+        // Find any serviceProviderIds that are not currently attached
+        const missingServiceProviderIds: string[] = serviceProviderIds.filter(
+            (id: string) => !this.serviceProviderIds.includes(id),
+        );
+
+        // If any IDs are missing, return an error indicating which ones were not found
+        if (missingServiceProviderIds.length > 0) {
+            return new EntityNotFoundError(
+                'Rolle ServiceProvider Verknüpfung',
+                `The following service-provider IDs were not found: ${missingServiceProviderIds.join(', ')}`,
+            );
+        }
+
+        // Filter out all the serviceProviderIds that need to be detached
+        this.serviceProviderIds = this.serviceProviderIds.filter((id: string) => !serviceProviderIds.includes(id));
+    }
+
+    public async updateServiceProviders(serviceProviderIds: string[]): Promise<void | DomainError> {
+        // Fetch current state of service providers from the DB (those already attached)
+        const existingServiceProviderIds: string[] = this.serviceProviderIds;
+
+        // Identify the service provider IDs to add (those in the new state but not in the current state)
+        const serviceProviderIdsToAdd: string[] = serviceProviderIds.filter(
+            (id: string) => !existingServiceProviderIds.includes(id),
+        );
+
+        // Identify the service provider IDs to remove (those in the current state but not in the new state)
+        const serviceProviderIdsToRemove: string[] = existingServiceProviderIds.filter(
+            (id: string) => !serviceProviderIds.includes(id),
+        );
+
+        // If there are IDs to add, call the attach method
+        if (serviceProviderIdsToAdd.length > 0) {
+            const result: void | DomainError = await this.attachServiceProviders(serviceProviderIdsToAdd);
+            if (result instanceof DomainError) {
+                return result;
+            }
+        }
+
+        // If there are IDs to remove, call the detach method
+        if (serviceProviderIdsToRemove.length > 0) {
+            const result: void | DomainError = this.detatchServiceProvider(serviceProviderIdsToRemove);
+            if (result instanceof DomainError) {
+                return result; // Return error if detachment failed
+            }
         }
```

Comment:
I mean, sure, it's a quicker fix, but it could lead to some issues down the line. We got to  make sure all those IDs actually exist in the first place and return an exception if that is the case (which we can't do if we just set whole array like that) and avoiding any duplicate entries. Plus, you don't wanna risk losing any of the current IDs in the process. 
The way I see it, it's better to break it down, figure out what needs to be added or removed, and handle each case carefully. That way, you can keep things clean and organized, without any unwanted surprises.

Reply:
Also with the Personenkontexte we did the same thing (Adding/deleting only when necessary)

Reply:
Could be but I went the other way to separate concerns and since the method detacheServiceProvider is already there, I thought I could just use it. This way is also a bit more consistent since it also adds/deletes whats necessary and exceptions are also thrown separately (We know what went wrong where exazctly).

But that much code isn't worth the prize and you are right :) Less code to maintain, less tests and in this case, same result.. so I refactored it.